### ISSUE TITLE
FEAT: In `skore.train_test_split`, shuffle should be always True

### ISSUE THREAD

**Issue by sylvaincom** [probabl-ai/skore#861]
In `skore.train_test_split`, `shuffle` should be `True` by default, and raise a warning only for time series!

**Comment by ogrisel** [comment]
And if a `TimeSeriesSplit` is used in `cross_validate`, one should warn the user if there is a single timestamp column and that the data is not ordered by this column. This should probably be turned in a dedicated issue/feature request and cross-linked to this one.

**Comment by adrinjalali** [comment]
when dealing with time, it's not at all clear how to split things. Sometimes times are attached to another entity like customer, instead of being one single thread of time, sometimes it's not the timestamp of the event, but something coming from another entity attached/joined with the main one being investigated.

So a clear recommendation when datetime is involved, is not really possible.

**Comment by sylvaincom** [comment]
- If Shuffle is set to True: don't raise any warnings.
- If Shuffle is set to False: raise a warning that also explains that we do not need to put it to True only if it is time-based.
- Raise a warning if we detect time-based features. (This covers the Shuffle=True but with time series case.)

WDYT @MarieS-WiMLDS?

Then, we can have a specific feature for:
>And if a TimeSeriesSplit is used in cross_validate, one should warn the user if there is a single timestamp column and that the data is not ordered by this column. This should probably be turned in a dedicated issue/feature request and cross-linked to this one.

**Comment by MarieS-WiMLDS** [comment]
I'm not so sure about it; when we first discussed it, we said that even if we don't detect time column, most of the situations are actually ordered in time, and this is why this warning was interesting because it triggers the user into thinking about their current situation. For instance in a system to detect out of service road sign. You don't care if it's on a Friday or Wednesday, Winter or Summer, but maybe more and more roadsigns will be added to the system in time, with their particularities. Doing shuffle will therefore erase the fact that the diversity of situations where the system is used grows.

**Comment by glemaitre** [comment]
I don't think that the problem is that simple.

If the ordering of the dataset is related to an underlying structure, it might make sense to not break this structure during the evaluation and potentially the evaluation based on `shuffle=True` will provide over-optimistic scores (usually there is some grouping structure going on).

My first intuition to detect such issue without knowledge about the dataset would be to run the train/score with and without shuffling and if shuffling lead to a significant increase of the statistical metric then a warning should be raised (or at least I would go further to understand why to I get an improve score).

The following notebook gives an example of such issue: https://inria.github.io/scikit-learn-mooc/python_scripts/cross_validation_grouping.html. In this particular example, one should not use shuffling. I can provide a similar example showing this effect on the penguins dataset (there is a kind of ordering) but in this case, shuffling make sense. That's why, the decision of shuffling or not is only reasonable once you identify the structure and the way forward. However, the warning is extremely valuable because it is rather tricky to debug.

**Comment by auguste-probabl** [comment]
May be outdated
